src/Randomizer.SMZ3.Tracking/TrackerServiceCollectionExtensions.cs

```diff
@@ -47,7 +47,7 @@ public static IServiceCollection AddTracker(this IServiceCollection services)
             services.AddScoped<PushToTalkSpeechRecognitionService>();
             services.AddScoped<NullSpeechRecognitionService>();
             services.AddTransient<IMicrophoneService, MicrophoneService>();
-            services.AddSingleton<IGlobalHook, TaskPoolGlobalHook>();
+            services.AddScoped<IGlobalHook, TaskPoolGlobalHook>();
```

Comment:
Why did you change this to scoped? The [library documentation](https://sharphook.tolik.io/v5.2.0/articles/hooks.html) specifically says to only have one instance active at a time in the application.

Reply:
Well, with the way this is setup and the fact you can only have a single tracker active, so it should only have one instance available at a time.

But, the issue was that it was crashing when closing tracker and re-opening because closing the prior tracker disposed of the hook. When I removed the dispose, it seemed to hang on closing tracker.

Maybe the dispose can be handled on closing the app, perhaps?

Reply:
Pushed in a fix for the id, but after messing with it a bit, I'm really just not familiar enough with this library you're using quickly discern best practice here. You have it trying to rerun the hook each time tracker is loaded, and if it's on a singleton that messes up on the second run even if I try to wait until app close to dispose of them.

I suppose I could have the hook and hook runner task initialized on the app launch, then the PushToTalkSpeechRecognitionService just binds/unbinds the events?

Reply:
Went ahead and did that. The hook and hook task are now initialized in the app and disposed of on exit. Tested, and I'm able to close tracker and reopen with no issues.